pages/models/generations/Sources/epri_gfm/index.md

```diff
@@ -0,0 +1,276 @@
+---
+layout: page 
+title: EPRI Grid Forming
+tags: ["Opensource", "RMS", "GFM", "GFL", "converter", "wind", "generic", "pv", "hvdc", "EPRI", "Dynawo"] 
+date: 20/12/2024
+last-updated: 20/12/2024
+id: #63
+authors: Nils Wiese (Fraunhofer IEE), Martin Franke (Fraunhofer IEE), Saikrishna Vallabhaneni
+reviewers: Lampros Papangelis (CRESYM)
+---
+
+## Context
+
+This model was developped by Electric Power Research Institute (EPRI)
+and provides a standarized approch for inverter based resources [1].
+The orignal model is described in a PDF available on the
+[webpage](https://www.epri.com/research/products/000000003002021403).
+
+## Model use, assumptions, validity domain and limitations
+
+The model is a positive-sequence RMS model, hence it assumes symmetrical
+operating conditions and neglects high-frequency dynamics. This type of
+model is often used in large-scale stability studies, for which it
+reflects the relevant phenomena. It is not a detailed physical model of
+the unit. Also for some stability phenomena (e.g. resonance stability)
+this model is not sufficient and EMT models or other approaches may be
+necessary. A comparision against EMT is shown in the EPRI documentation
+. It is also available for
+[PSCAD](https://www.pscad.com/knowledge-base/download/PSCAD-EPRI-Generic-GFM.pdf&ved=2ahUKEwjOpr_z0rOKAxVPg_0HHUO_JAgQFnoECBoQAQ&usg=AOvVaw1LVPVn2jXqTdzbFxeCoPo7).
+
+## Model description
+
+### Difference between grid following and grid forming inverter
+
+A grid-following inverter synchronizes itself with the voltage and
+frequency values of the power grid using a Phase-Locked Loop (PLL). The
+PLL tracks the phase angle of the grid and therefore enables the
+inverter control to follow the grid. In contrast, a grid-forming
+inverter actively controls the grid’s voltage and frequency, enabling it
+to create or stabilize the grid, especially in scenarios where
+traditional grid infrastructure is weak or absent. Grid-forming
+inverters do not rely on a PLL. Many approaches synchronize with the
+grid through the power like a synchronous generator.
+
+### EPRI GFM operation modes
+
+The Generic EPRI GFM model offers four control modes:
+
+- Droop based, $$\omega_\mathrm{flag}=1$$, see
+  <a href="#fig-droop_mode" class="quarto-xref">Figure 1</a>
+- Virtual Synchronous Machine (VSM), $$\omega_\mathrm{flag}=2$$, see
+  <a href="#fig-vsm_mode" class="quarto-xref">Figure 2</a>
+- Dispatchable Virtual Oscillator (dVOC) based GFM,
+  $$\omega_\mathrm{flag}=3$$, see
+  <a href="#fig-dvoc_mode" class="quarto-xref">Figure 3</a>
+- Phase Locked Loop (Grid following mode), $$\omega_\mathrm{flag}=0$$, see
+  <a href="#fig-pll_mode" class="quarto-xref">Figure 4</a>
+
+## Model schema
+
+The different control method diagrams of the GFM is shown below:
+
+
+![](./drawings/Droop_Mode.drawio.svg)
```

Comment:
There is a sign problem in the calculation of Ed or Eq (depending on the convention). It should be either +i_d X_f or +i_q X_f